pico/memory.c

```diff
@@ -1431,7 +1431,8 @@ static int ym2612_write_local(u32 a, u32 d, int is_from_z80)
           elprintf(EL_YMTIMER, "st mode %02x", d);
           ym2612_sync_timers(cycles, old_mode, d);
 
-          ym2612.OPN.ST.mode = d;
+          if ((d ^ old_mode) & 0xc0)

```

Comment:
Now I have the impression that everything in this case branch but for the sync_timers is superfluous, since all of the rest is happening in Write, or am I overlooking something?

Reply:
That makes sense. Sorry for having been a bit dense, I should have seen this myself :-/